src/dbnode/persist/fs/commitlog/chunk_reader.go

```diff
@@ -80,18 +83,36 @@ func (r *chunkReader) readHeader() error {
 		return err
 	}
 
-	// Verify data checksum
-	data, err := r.buffer.Peek(int(size))
+	// Setup a chunk data buffer so that chunk data can be loaded into it.
+	chunkDataSize := int(size)
+	if chunkDataSize > cap(r.chunkData) {
+		// Increase chunkData capacity so that it can fit the new chunkData.
+		chunkDataCap := cap(r.chunkData)
+		for chunkDataCap < chunkDataSize {
+			chunkDataCap *= 2
+		}
+		r.chunkData = make([]byte, chunkDataSize, chunkDataCap)
+	} else {
+		//Reuse existing chunk data buffer if possible
```

Comment:
nit: Add space at start and "." to this comment.
```
// Reuse existing chunk data buffer if possible.
```

Reply:
Fixed.